src/docs/sphinx/tutorials/step12/Tutorial.rst

```diff
@@ -0,0 +1,357 @@
+.. _TutorialPoroPlasticity:
+
+
+####################################################
+Tutorial 12: PoroMechanical Models for Wellbore Problems
+####################################################
+
+
+**Context**
+
+The main objective of this tutorial is to demonstrate how to use the internal wellbore mesh generator and poromechanical solvers in GEOSX to tackle wellbore problems in porous media. In this tutorial, a poroplastic model is applied to find the solution of rock deformation within the vicinity of a vertical wellbore, considering elastoplastic deformation, fluid diffusion and poromechanical coupling effect. To do so, a single phase flow solver is fully coupled with a Lagrangian mechanics solver in GEOSX and the Extended Drucker-Prager model (see :ref:`TwoInvariantPlasticity`) is chosen as the material model for the solid domain. We first solve this problem with the existing poroelastic solver and verify the modeling results with the corresponding analytical solutions. Then, the verified case is employed to test the newly implemented poroplasticity solver in GEOSX, whose results are compared with the ones obtained from the poroelastic case to highlight the impact of plasticity in this specific problem.
+ 
+
+**Objectives**
+
+At the end of this tutorial you will know:
+
+  - how to construct meshes for wellbore problems with the internal wellbore mesh generator,
+  - how to specify initial and boundary conditions, such as reservoir properties, in-situ stresses, mixed loading (mechanical and fluid) at wellbore wall and far-field constraints,
+  - how to use multiple solvers in GEOSX for predicting poroplastic deformations in the near wellbore region.
+
+
+**Input file**
+
+This tutorial uses no external input files and everything required is
+contained within a single GEOSX input file. 
+
+The xml input file for the test case with poroelasticity is located at:
+
+.. code-block:: console
+
+  examples/plasticity/WellboreProblem_PoroElastic.xml
+
+
+The xml input file for the test case with poroplasticity is located at:
+
+.. code-block:: console
+
+  examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+
+
+------------------------------------------------------------------
+Description of the case
+------------------------------------------------------------------
+
```

Comment:
General comments:

- It is not necessary to ever say "in GEOSX."  I would be impressed if the user was unaware of the code they are working in :-)
- We do not have separate Poroelastic and Poroplastic *solvers*.  These are one solver with different model specifications.  I would refer to them always as the "poroelastic version" and "poroplastic version" of the problem.